gh-135557: Use atomic writes on heapq operations (#135601)
python/cpython: Misc/NEWS.d/next/Library/2025-06-17-23-13-56.gh-issue-135557.Bfcy4v.rst

> Fix races on ``heapq`` updates and lock-free list reads in free-threaded

Reviewer: Refs:
```suggestion
Fix races on :mod:`heapq` updates and :class:`list` reads on the :term:`free threaded <free threading>`
```
